cmd/canary/canary-config/list_test.go

```diff
@@ -0,0 +1,168 @@
+	rootCmd.PersistentFlags().String("gate-endpoint", "", "Gate (API server) endpoint. Default http://localhost:8084")
+	rootCmd.PersistentFlags().Bool("insecure", false, "Ignore Certificate Errors")
+	rootCmd.PersistentFlags().Bool("quiet", false, "Squelch non-essential output")
+	rootCmd.PersistentFlags().Bool("no-color", false, "Disable color")
+	rootCmd.PersistentFlags().String("output", "", "Configure output formatting")
+	rootCmd.PersistentFlags().String("default-headers", "", "Configure additional headers for gate client requests")
+	util.InitUI(false, false, "")
+	return rootCmd
+}
+
+func TestCanaryConfigList_basic(t *testing.T) {
+	ts := testGateCanaryConfigListSuccess()
+	defer ts.Close()
+
+	// Exclude 'canary' since we are testing only the 'canary-config' subcommand.
+	args := []string{"canary-config", "list", "--gate-endpoint", ts.URL}
+	currentCmd := NewListCmd(canaryConfigOptions{})
+	rootCmd := getRootCmdForTest()
+	canaryConfigCmd := NewCanaryConfigCmd(os.Stdout)
+	canaryConfigCmd.AddCommand(currentCmd)
+	rootCmd.AddCommand(canaryConfigCmd)
+
+	rootCmd.SetArgs(args)
+	err := rootCmd.Execute()
+	if err != nil {
+		t.Fatalf("Command failed with: %s", err)
+	}
+}
+
+func TestCanaryConfigList_malformed(t *testing.T) {
+	ts := testGateCanaryConfigListMalformed()
+	defer ts.Close()
+
+	// Exclude 'canary' since we are testing only the 'canary-config' subcommand.
+	args := []string{"canary-config", "list", "--gate-endpoint", ts.URL}
+	currentCmd := NewListCmd(canaryConfigOptions{})
+	rootCmd := getRootCmdForTest()
+	canaryConfigCmd := NewCanaryConfigCmd(os.Stdout)
+	canaryConfigCmd.AddCommand(currentCmd)
+	rootCmd.AddCommand(canaryConfigCmd)
+
+	rootCmd.SetArgs(args)
+	err := rootCmd.Execute()
+	if err == nil {
+		t.Fatalf("Command failed with: %s", err)
+	}
+}
+
+func TestCanaryConfigList_fail(t *testing.T) {
+	ts := GateServerFail()
+	defer ts.Close()
+
+	// Exclude 'canary' since we are testing only the 'canary-config' subcommand.
+	args := []string{"canary-config", "list", "--gate-endpoint", ts.URL}
+	currentCmd := NewListCmd(canaryConfigOptions{})
+	rootCmd := getRootCmdForTest()
+	canaryConfigCmd := NewCanaryConfigCmd(os.Stdout)
+	canaryConfigCmd.AddCommand(currentCmd)
+	rootCmd.AddCommand(canaryConfigCmd)
+
+	rootCmd.SetArgs(args)
+	err := rootCmd.Execute()
+	if err == nil {
+		t.Fatalf("Command failed with: %s", err)
+	}
+}
+
+// testGateCanaryConfigListSuccess spins up a local http server that we will configure the GateClient
+// to direct requests to. Responds with a 200 and a well-formed canaryConfig list.
+func testGateCanaryConfigListSuccess() *httptest.Server {
+	mux := http.NewServeMux()
+	mux.Handle(
+		"/v2/canaryConfig",
+		http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			fmt.Fprintln(w, strings.TrimSpace(canaryConfigListJson))
+		}))
+	// Return a Gate version.
+	mux.Handle("/version", http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
```

Comment:
If you rebase and replace lines 101 and 116 with `mux := util.TestGateMuxWithVersionHandler()` this Handle call for version can be removed, also the one on 123.